File: kotlin-ide/j2k/old/testData/fileOrElement/nullability/nullableInitializer2.java
import org.jetbrains.annotations.Nullable;

public class TestJava {
    public Object nullableObj(int p) {
        return p > 0 ? "response" : null;
    }

    public String nullableInitializerFieldCast = (String) nullableObj(3);
    private String nullableInitializerPrivateFieldCast = (String) nullableObj(3);

    public void testProperty() {
        nullableInitializerFieldCast.charAt(0);
        nullableInitializerPrivateFieldCast.charAt(0);
    }

    public void testLocalVariable() {
        String nullableInitializerValCast = (String) nullableObj(3);

        nullableInitializerValCast.charAt(0);
    }
}

File: nj2k/testData/newJ2k/nullability/nullableInitializer2.kt
class TestJava {
    fun nullableObj(p: Int): Any? {
        return if (p > 0) "response" else null
    }

    var nullableInitializerFieldCast = nullableObj(3) as String?
    private val nullableInitializerPrivateFieldCast = nullableObj(3) as String?
    fun testProperty() {
        nullableInitializerFieldCast!![0]
        nullableInitializerPrivateFieldCast!![0]
    }

    fun testLocalVariable() {
        val nullableInitializerValCast = nullableObj(3) as String?
        nullableInitializerValCast!![0]
    }
}